server/worker/jira.go

```diff
@@ -265,15 +266,20 @@ func (j *Jira) Run(ctx context.Context, argsJSON json.RawMessage) error {
 func (j *Jira) runVuln(ctx context.Context, cli JiraClient, args jiraArgs) error {
 	vargs := args.Vulnerability
 	if vargs == nil {
-		// support the old format of vulnerability args, where only the CVE
-		// is provided.
-		vargs = &vulnArgs{
-			CVE: args.CVE,
-		}
+		return errors.New("invalid job args")
+	}
+
+	var hosts []fleet.HostVulnerabilitySummary
+	var err error
+
+	// Default to deprecated method in case we are processing an 'old' job payload
```

Comment:
Could you explain this change from using `HostsByCVE` to `HostVulnSummariesBySoftwareIDs`? (is the new process generating a job with the affected software IDs already calculated?)

Reply:
Ah, seems to be an optimization, given that we have the list of all `software_id`s here we pass it to the jobs, right?

Reply:
If so, let's document that the deprecation is for efficiency reasons.

Reply:
You got it